Twits/Controllers/TwitController.cs

```diff
@@ -0,0 +1,214 @@
+                SqlDataReader reader = command.ExecuteReader();
+                while (reader.Read())
+                {
+                    Twit twit = new Twit();
+                    twit.Id = reader.GetInt32(0);
+                    twit.Text = reader.GetString(1);
+                    twit.CreatedOn = reader.GetDateTimeOffset(2);
+                    twits.Add(twit);
+                }
+            }
+
+            return View(twits);
+        }
+
+        //// GET: Twit/Details/5
+        //public ActionResult Details(int id)
+        //{
+        //    return View();
+        //}
+
+        //// GET: Twit/Create
+        public ActionResult Create()
+        {
+            Twit twit = new Twit();
+            return View(twit);
+        }
+
+        //// POST: Twit/Create
+        [HttpPost]
+        public ActionResult Create(FormCollection collection)
+        {
+            Twit twit = new Twit();
+            twit.Text = collection.Get("Text");
+            twit.CreatedOn = DateTimeOffset.Now;
+            try
+            {
+                string connectionString = ConfigurationManager.ConnectionStrings["TwitDatabase"].ConnectionString;
+
+                using (SqlConnection connection = new SqlConnection(connectionString))
+                {
+                    connection.Open();
+                    string sql = @"
+                        INSERT INTO Twits(Text, CreatedOn)
+                        VALUES (@text, @createdOn)
+                    ";
+                    SqlCommand command = new SqlCommand(sql, connection);
+                    command.Parameters.AddWithValue("@text", twit.Text);
+                    command.Parameters.AddWithValue("@createdOn", twit.CreatedOn);
+                    command.ExecuteNonQuery();
+                }
+
+                return RedirectToAction("Index");
+            }
+            catch
+            {
+                return View(twit);
+            }
+        }
+
+        public ActionResult Display()
+        {
+            string connectionString = ConfigurationManager.ConnectionStrings["TwitDatabase"].ConnectionString;
+
+            var model = new List<Twit>();
+
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+                string sql = @"SELECT Text, CreatedOn FROM Twits
+                    Order by CreatedOn ASC";
+                SqlCommand command = new SqlCommand(sql, connection);
+                SqlDataReader rdr = command.ExecuteReader();
+                while (rdr.Read())
+                {
+                    var twit = new Twit();
+                    twit.Text = rdr["Text"].ToString();
+                    string dateFormat = "MM/dd/yyyy hh:mm";
+                    DateTimeOffset date = (DateTimeOffset)rdr["CreatedOn"];
+                    string formattedDate = date.ToString(dateFormat);
+                    twit.CreatedOn = Convert.ToDateTime(formattedDate);
```

Comment:
I'd avoid converting the `DateTimeOffset` to a `DateTime`. To format the `CreatedOn` value, simply call the `ToString` method on that property in the view and pass in your format string:

```
@twit.CreatedOn.ToString("MM/dd/yyyy hh:mm")
```